python/ray/_private/worker.py

```diff
@@ -2519,9 +2525,11 @@ def get(
             blocking_get_inside_async_warned = True
 
     with profiling.profile("ray.get"):
-        # TODO(sang): Should make StreamingObjectRefGenerator
-        # compatible to ray.get for dataset.
         if isinstance(object_refs, StreamingObjectRefGenerator):
+            # TODO(sang): We should raise an exception.
```

Comment:
Raising an exception requires lots of test changes. I will fix it in the third PR.